Add the ability for CssBuilder to properly Add multiple classes without duplicates

StevenTCramer opened the pull request:
<!---
Thanks for filing a pull request 😄 ! Before you submit, please read the following:

Search open/closed issues before submitting. Someone may have pushed the same thing before!

Provide a summary of your changes in the title field above.
-->

# Pull Request

## 📖 Description

1. **Adding AddClasses Method:**
   - Introduced `AddClasses` to handle multiple space-separated CSS classes efficiently, leveraging the `HashSet` to eliminate duplicates.

2. **Refactoring AddClass Methods:**
   - Updated `AddClass` methods to internally call `AddClasses`, ensuring backward compatibility.

3. **Conditional AddClasses Methods:**
   - Added conditional `AddClasses` methods to add classes based on boolean conditions or functions.

These changes optimize class addition by preventing duplicates and improving performance.

<!---
Provide some background and a description of your work.
What problem does this change solve?
Is this a breaking change, chore, fix, feature, etc?
-->

### 🎫 Issues

The current AddClass method actually allows for adding multiple but it doesn't split the input and utilize the hashset and thus would allow duplicates.

<!---
* List and link relevant issues here.
-->

## 👩‍💻 Reviewer Notes

<!---
Provide some notes for reviewers to help them provide targeted feedback and testing.

Do you recommend a smoke test for this PR? What steps should be followed?
Are there particular areas of the code the reviewer should focus on?
-->

## 📑 Test Plan

Update CSSBuilder Tests cases to cover 

<!---
Please provide a summary of the tests affected by this work and any unique strategies employed in testing the features/fixes.
-->

## ✅ Checklist

### General

<!--- Review the list and put an x in the boxes that apply. -->

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fluentui-blazor/blob/master/docs/contributing.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

<!--- Review the list and put an x in the boxes that apply. -->
<!--- Remove this section if not applicable. -->

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [ ] I have modified an existing component
- [ ] I have validated the [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component 

## ⏭ Next Steps

<!---
If there is relevant follow-up work to this PR, please list any existing issues or provide brief descriptions of what you would like to do next.
-->

dvoituron commented:
This could be an interesting improvement. But why not keep the existing ˋAddClass` method, improving the internal code? And don't add new methods that don't solve the current usage.

The CSS class name must be composed of numbers, hyphens, underscores and letters.

So the existing ˋAddClass` method could be split into spaces, commas, ... to use your code.

StevenTCramer commented:
* Update AddClass to support single or multiple classes.  
* Ignore invalid classes based on the regex. 
* Updated tests cases to validate.
*  Fixed StyleBuilder Tests copy and paste errors.

dvoituron commented on tests/Core/Utilities/StyleBuilderTests.cs:
```diff
@@ -71,20 +71,6 @@ public void CssBuilder_WithComplexUserStyle()
         Assert.Equal("color: red; font-size: 12px; font-name: courier;", styleBuilder.Build());
     }
 
-    [Fact]
```
You can perhaps keep (and correct as follows) this (additional) test to check the use of several classes.

```csharp
[Fact]
public void CssBuilder_WithComplexUserClasses()
{
    // Assert
    var cssBuilder = new CssBuilder("   my-user-class1     my-user-class2  ");

    // Act
    cssBuilder.AddClass("  class1  ");
    cssBuilder.AddClass("  class2  ");

    // Assert - Values are sorted
    Assert.Equal("class1 class2 my-user-class1 my-user-class2", cssBuilder.Build());
}
```

StevenTCramer replied:
That was a CssBuilder test in the `StyleBuilderTests.cs` file.  And we have similar tests

CssBuilder_AddMultipleValidUserClasses
CssBuilder_AddClassesWithExtraSpaces

In the appropriate CssBuilderTests.cs file.

But I did see I was missing a few  test cases.  I added.

CssBuilder_AddMultipleClassesAtOnce
CssBuilder_AddMultipleClassesAtOnceWithUserClass
CssBuilder_AddMultipleClassesAtOnceWithExtraSpaces

dvoituron reviewed (changes requested):
Good work. I like it :-)

azunyuuuuuuu commented:
This commit sadly breaks using Tailwind CSS with this library. Classes like `hover:bg-red-400`, `grid-cols-[1fr_3fr]` or `[&>a]:text-sm` get eaten up by the regex.

Is there a way to disable or change the behaviour of the validation?

vnbaaij commented:
No, currently there is no way to prevent that from happening.

dvoituron commented:
> This commit sadly breaks using Tailwind CSS with this library. Classes like `hover:bg-red-400`, `grid-cols-[1fr_3fr]` or `[&>a]:text-sm` get eaten up by the regex.

Weird. These strings are not valid class names.

> a name must begin with an underscore (_), a hyphen (-), or a letter(a–z), followed by any number of hyphens, underscores, letters, or numbers.
https://stackoverflow.com/questions/448981/which-characters-are-valid-in-css-class-names-selectors
